### ISSUE TITLE
Support Redis

### ISSUE THREAD

**Issue by mathstuf** [ejfitzgerald/clang-tidy-cache#5]
Like #2 for memcached, except we use Redis to store cache artifacts across all of our CI machines.

**Comment by ejfitzgerald** [comment]
I am thinking about adding a redis backend as you suggest. Would you be interested/available to do some beta testing if I do?

**Comment by mathstuf** [comment]
Yeah, I can find some time for that. `clang-tidy` is by *far* our bottleneck in CI when hardware is otherwise available :) .
